keanu-project/src/main/java/io/improbable/keanu/distributions/discrete/Poisson.java

```diff
@@ -78,27 +77,10 @@ private static int sample(double mu, KeanuRandom random) {
 
     @Override
     public DoubleTensor logProb(IntegerTensor k) {
-        Tensor.FlattenedView<Double> muFlattenedView = mu.getFlattenedView();
-        Tensor.FlattenedView<Integer> kFlattenedView = k.getFlattenedView();
-
-        double[] result = new double[(int) k.getLength()];
-        for (int i = 0; i < result.length; i++) {
-            result[i] = Math.log(pmf(muFlattenedView.getOrScalar(i), kFlattenedView.getOrScalar(i)));
-        }
 
-        return DoubleTensor.create(result, k.getShape());
-    }
+        DoubleTensor kDouble = k.toDouble();
+        DoubleTensor logFactorialK = kDouble.plus(1).applyInPlace(Gamma::logGamma);
```

Comment:
shouldn't this be `.logGammaInPlace()` instead of `.applyInPlace(Gamma::logGamma)`